.github/ci-scripts/bash/functions-for-build

```diff
@@ -263,6 +269,26 @@ frp_build_and_push() {
   set +e
 }
 
+gcp_docker_build() {
+  set -e
+  IMAGE="${1}"
+  download_go_libraries
+  docker_build "$@"
+  set +e
+}
+
+gcp_docker_build_and_push() {
+  IMAGE="${1}"
+  download_go_libraries
+  login_to_vault
```

Comment:
For dockerhub login, we use a github-action. The same github-action supports google cloud also.  Check if we can use the same?  If yes, then login into docker registry will become conditional based on `APPLICATION_NAME` variable value.

For docker login - Check if we can use this - https://github.com/docker/login-action?tab=readme-ov-file#google-container-registry-gcr